tests/tests/test_state_scripts.py

```diff
@@ -873,6 +873,147 @@ def fetch_info(cmd_list):
                 % (client_service_name, client_service_name)
             )
 
+    TEST_SET_ROLLBACK = (
+        "Normal_success",
+        {
+            "FailureScript": "ArtifactInstall_Leave_89",
+            "ScriptOrder": [
+                "Idle_Enter_08_testing",
+                "Idle_Enter_09",
+                "Idle_Leave_09",
+                "Idle_Leave_10",
+                "Sync_Enter_02",
+                "Sync_Enter_03",
+                "Sync_Leave_04",
+                "Sync_Leave_15",
+                "Download_Enter_12",
+                "Download_Enter_13",
+                "Download_Leave_14",
+                "Download_Leave_25",
+                "ArtifactInstall_Enter_01",
+                "ArtifactInstall_Enter_02",
+                "ArtifactInstall_Leave_01",
+                "ArtifactInstall_Leave_03",
+                "ArtifactRollback_Enter_01",
+                "ArtifactRollback_Enter_11",
+                "ArtifactRollback_Leave_01",
+                "ArtifactRollback_Leave_89",
+                "ArtifactFailure_Enter_01",
+                "ArtifactFailure_Leave_06",
+                "Idle_Enter_08_testing",  # Update attempt finished
+                "Idle_Enter_09",
```

Comment:
Great!

But do you need a new test for this? Can't you just add this as one more case to the existing test?

Reply:
I wonder the same, but it seems like the logic is a bit different.

In any case, for consistency I suggest we move `TEST_SET_ROLLBACK` to before the class definition, just after `REBOOT_TEST_SET`, or (better?) we move`TEST_SETS` and `REBOOT_TEST_SET` to just above the test methods that use them.